src/farcasterd/runtime.rs

```diff
@@ -918,17 +424,22 @@ impl Runtime {
                     ServiceBus::Ctl,
                     ServiceId::Farcasterd, // source
                     source,                // destination
-                    Request::SwapList(self.running_swaps.iter().cloned().collect()),
+                    Request::SwapList(
+                        self.trade_state_machines
+                            .iter()
+                            .filter_map(|tsm| tsm.swap_id())
+                            .collect(),
+                    ),
                 )?;
             }
 
             Request::ListOffers(offer_status_selector) => {
                 match offer_status_selector {
                     OfferStatusSelector::Open => {
-                        let pub_offers = self
-                            .public_offers
+                        let open_offers = self
+                            .trade_state_machines
                             .iter()
-                            .filter(|k| !self.consumed_offers_contains(k))
+                            .filter_map(|tsm| tsm.open_offer())
```

Comment:
incidentally, also more efficient for a long-running node with large history of swaps